hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2646,52 +2646,99 @@ func reconcileCAPIManagerClusterRoleBinding(binding *rbacv1.ClusterRoleBinding,
 	return nil
 }
 
-func reconcileCAPIManagerRole(role *rbacv1.Role) error {
-	role.Rules = []rbacv1.PolicyRule{
-		{
-			APIGroups: []string{
-				"bootstrap.cluster.x-k8s.io",
-				"controlplane.cluster.x-k8s.io",
-				"infrastructure.cluster.x-k8s.io",
-				"machines.cluster.x-k8s.io",
-				"exp.infrastructure.cluster.x-k8s.io",
-				"addons.cluster.x-k8s.io",
-				"exp.cluster.x-k8s.io",
-				"cluster.x-k8s.io",
+func reconcileCAPIManagerRole(role *rbacv1.Role, platformType hyperv1.PlatformType) error {
+	if platformType == hyperv1.IBMCloudPlatform {
+		role.Rules = []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{
+					"bootstrap.cluster.x-k8s.io",
+					"controlplane.cluster.x-k8s.io",
+					"infrastructure.cluster.x-k8s.io",
+					"machines.cluster.x-k8s.io",
+					"exp.infrastructure.cluster.x-k8s.io",
+					"addons.cluster.x-k8s.io",
+					"exp.cluster.x-k8s.io",
+					"cluster.x-k8s.io",
+				},
+				Resources: []string{"*"},
+				Verbs:     []string{"*"},
 			},
-			Resources: []string{"*"},
-			Verbs:     []string{"*"},
-		},
-		{
-			APIGroups: []string{"hypershift.openshift.io"},
-			Resources: []string{
-				"hostedcontrolplanes",
-				"hostedcontrolplanes/status",
+			{
+				APIGroups: []string{"hypershift.openshift.io"},
+				Resources: []string{
+					"hostedcontrolplanes",
+					"hostedcontrolplanes/status",
+				},
+				Verbs: []string{"get", "list", "watch"},
 			},
-			Verbs: []string{"*"},
-		},
-		{
-			APIGroups: []string{""},
-			Resources: []string{
-				"configmaps",
-				"events",
-				"nodes",
-				"secrets",
+			{
+				APIGroups: []string{""},
+				Resources: []string{
+					"configmaps",
+					"secrets",
+				},
+				Verbs: []string{"list", "watch"},
 			},
-			Verbs: []string{"*"},
-		},
-		{
-			APIGroups: []string{"coordination.k8s.io"},
-			Resources: []string{
-				"leases",
+			{
+				APIGroups: []string{"coordination.k8s.io"},
+				Resources: []string{
+					"leases",
+				},
+				Verbs: []string{"*"},
 			},
-			Verbs: []string{"*"},
-		},
-		{
-			APIGroups: []string{"capi-provider.agent-install.openshift.io"},
-			Resources: []string{"*"},
-			Verbs:     []string{"*"},
-		},
+			{
```

Comment:
why do you need the one for agent?

Reply:
@enxebre I'll check with Tyler, but I'm guessing it was just to stay consistent with what was already there?